template.go

```diff
@@ -9,65 +9,41 @@ import (
 )
 
 func (c *Client) GetTemplate(code string) (*objects.Template, error) {
-	res, err := c.request(&request{
-		method:      http.MethodGet,
-		path:        fmt.Sprintf(TemplateFmt, code),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
 	template := &objects.Template{}
-	if err = res.JSON(template); err != nil {
-		return nil, err
-	}
-
-	return template, nil
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(TemplateFmt, code)).
+		Expect(http.StatusOK).
+		ContentType(JsonContentType).
+		Bind(template).
+		Send(c)
+
+	return template, err
 }
 
 func (c *Client) CreateGuildFromTemplate(code string, reason string) (*objects.Guild, error) {
-	res, err := c.request(&request{
-		method:      http.MethodPost,
-		path:        fmt.Sprintf(TemplateFmt, code),
-		contentType: JsonContentType,
-		reason:      reason,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	if err = res.ExpectsStatus(http.StatusOK); err != nil {
-		return nil, err
-	}
-
 	guild := &objects.Guild{}
-	if err = res.JSON(guild); err != nil {
-		return nil, err
-	}
-	return guild, nil
+	err := NewRequest().
+		Method(http.MethodPost).
+		Path(fmt.Sprintf(TemplateFmt, code)).
+		Expect(http.StatusOK).
+		ContentType(JsonContentType).
+		Bind(guild).
+		Reason(reason).
+		Send(c)
+	return guild, err
 }
 
 func (c *Client) GetGuildTemplates(guild objects.Snowflake) ([]*objects.Template, error) {
-	res, err := c.request(&request{
-		method:      http.MethodGet,
-		path:        fmt.Sprintf(GuildTemplateFmt, guild),
-		contentType: JsonContentType,
-	})
-	if err != nil {
-		return nil, err
-	}
-
-	var templates []*objects.Template
-	if err = res.JSON(&templates); err != nil {
-		return nil, err
-	}
-
-	return templates, nil
+	templates := []*objects.Template{}
+	err := NewRequest().
+		Method(http.MethodGet).
+		Path(fmt.Sprintf(GuildTemplateFmt, guild)).
+		Expect(http.StatusOK).
+		ContentType(JsonContentType).
+		Bind(templates).
```

Comment:
We need to pass a pointer to the slice here, otherwise it will automatically return an error.